lib/harmony/harmonytask/task_type_handler.go

```diff
@@ -116,6 +118,15 @@ top:
 		}
 	}
 
+	var location string
+	if c := h.TaskTypeDetails.Cost.Storage.Claim; c != nil {
+		if location, err = c(int(*tID)); err != nil {
```

Comment:
This won't work nicely with paths shared between nodes:
* Start 100 tasks
* 10 machines try to claim on one path, let's say 2 succeed, rest will fail the task they have claimed